src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -539,9 +547,9 @@ class _DataHookSelector:
         datamodule: A LightningDataModule
     """
 
-    model: "pl.LightningModule"
+    model: Optional["pl.LightningModule"]
     datamodule: Optional["pl.LightningDataModule"]
-    _valid_hooks: Tuple[str] = field(
+    _valid_hooks: Tuple[str, str, str] = field(
```

Comment:
I don't think that limiting `_valid_hooks` to length 3 is a proper approach. It makes sense to make it variable length (https://docs.python.org/3/library/typing.html#typing.Tuple)

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13806/commits/6020d0b3ec9afa56ee34757adea889163eac16bc